### ISSUE TITLE
WOW YOU REALLY ARE SUPER, @SuperJMN!! :) :) :)

### ISSUE THREAD

**Issue by Mike-EEE** [OmniGUI/OmniGUI#2]
WOWWWW.... this is a ping-ping to your pings at https://github.com/Microsoft/xaml-standard/issues/20#issuecomment-310050289 and https://github.com/Microsoft/xaml-standard/issues/193#issuecomment-310050760 and thought I would post in here instead as it tends to get a little stuffy in there. 😄 😆 

I've also sent out a little ping to the [Create a Ubiquitous .NET Client Application Development Model](https://visualstudio.uservoice.com/forums/121579-visual-studio-2015/suggestions/10027638-create-a-ubiquitous-net-client-application-develo) vote as that has turned into a semi-news hub of .NET client models (feel free to post a note there too if you feel so inclined).

So what's the scoop yo?  Totally doing your own thing here?  I am in complete agreement with your approach, of course.  As you know, [a consistent user experience is what I consider a necessary ubiquitous quality](http://blog.developers.win/2015/10/ubiquitous-qualities/#consistent-user-experience) and one that [Xamarin.Forms](http://blog.developers.win/2015/10/existing-net-client-application-models/#xamarinforms) -- as impressive as it is -- takes a 💩 all over, LOL.

How does this square with Avalonia?  Supplements it?  Totally your own thing now?  Details please. :) :) :)

(Also, @birbilis... looks like ANOTHER entry for my [list of client models](http://blog.developers.win/2015/10/existing-net-client-application-models/). Sigh.  Gotta brush the dust off that thing. 🤕)

**Comment by Mike-EEE** [comment]
OK @SuperJMN I got some <3 for you here:
http://blog.developers.win/2015/10/existing-net-client-application-models/#omnigui

(Unfortunately the quality matrix at the bottom requires an FTP so that is not updated just yet and probably won't be for a while.)

**Comment by SuperJMN** [comment]
Oh my God! You alone in one day have achieved more notoriety than me in a year 😮 

Big, big thanks, Mike! by the way, I must say I would like to have some conversation with you, with some messenger program. You are one of the most enthusiastic people I've ever watched (because haven't met, yet!)

**Comment by SuperJMN** [comment]
Mike, yes, I'm doing it all on my own. I already told some colleagues about, but it seems I'm the only one interested in something like this 😋 BTW, I have pinged @migueldeicaza (https://github.com/migueldeicaza) and he seemed to like it! 

Also, I won't bother @Grokys because Avalonia has already migrated to Portable.Xaml and it's a pretty mature UI framework with a lot of collaborators. I understand they had to switch from OmniXAML to Portable.Xaml because they expect a XAML parser that works, with all the features packed, something that OmniXAML cannot provide just yet).

**Comment by SuperJMN** [comment]
Just to clarify things a bit about the project: 
OmniGUI sits over simplicity. It's thought to be a component that fits into existing applications. In each platform it supports, OmniGUI will render inside a UIView, View, Control... or whatever a "UI component" is called in the target platform. 

It will take the client area assigned to it and render like any other component, using drawing primitives available in the platform.

The interaction with it is also made possible because there are abstraction layers that provide the signals OmniGUI needs to make its components act in consequence to user actions.

For drawing, in iOS it uses CoreGraphics, in Android it uses Canvas, in UWP it uses Win2D and in WPF it uses a DrawingContext.

**Comment by Mike-EEE** [comment]
Hehehe... yeah I find there's not a lot of luv in the trenches.  So I will admit I sort of add to it a bit to make up for it.  It's a cold world out there, what can I say. 👼 

That's great to hear about Portable.Xaml and Avalonia.  I have nothing but respect for that team and vision.  They are also "with it."  XF was a good stab, but I just don't see a lot of good things about it unless I am reading a Xamarin blog post. ;P

I haven't forgot about [your IM request](https://github.com/wojtpl2/ExtendedXmlSerializer/issues/101#issuecomment-300766878)!  I was going to get around to it when I started to hash all the (growing) issues in our repo at ExtendedXmlSerializer.  It would be awesome to team up in some way with Omni*.  I've been thinking about it. :)  I will add you on Skype now, my man.  Thanks again for being a passionate visionary. 👍

**Comment by grokys** [comment]
@SuperJMN just for the record, we've not switched to Portable.Xaml yet, though we hope to soon!

This looks really interesting, I will be keeping an eye on it! ;)

**Comment by SuperJMN** [comment]
Hey @grokys!! Welcome! Thanks for coming 😄 I'm really sorry because OmniXAML couldn't keep the track of AvaloniaUI while it started to gain traction. Now it rocketed and needs to fly free! Well, it seems that my beloved XAML parser was too basic and intricate. I redesigned it n-times to face some big challenges. I hope Portable.Xaml is a booster for your project, instead of a flat tire.

BTW, I created OmniGUI just as proof-of-concept, to learn what I always wanted to learn, and I'm actually enjoying it. 

What should I say? Without your inspiration and Avalonia it's very unlikely that this would have ever existed. I had a (very) bad time seeing that my real-life job took away my chances to be an important part in your project and finally understood that I had lost Avalonia's train.

I don't know where this will lead, I don't have high hopes, but thanks for everything.
